localstack/cli/localstack.py

```diff
@@ -403,6 +404,9 @@ def cmd_start(docker: bool, host: bool, no_banner: bool, detached: bool) -> None
         config.is_in_docker = False
         config.dirs = config.init_directories()
 
+        # call hooks to prepare host (note that this call should stay below the config overrides above)
```

Comment:
This is effectively the main change in the PR - calling the hooks _after_ we've adjusted the `config` values above.